drivers/health_pctp/app/controllers/health_pctp/careplans_controller.rb

```diff
@@ -0,0 +1,74 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HealthPctp
+  class CareplansController < IndividualPatientController
+    include AjaxModalRails::Controller
+
+    before_action :set_client
+    before_action :set_hpc_patient
+    before_action :set_careplan, only: [:edit, :update, :show, :destroy]
+
+    def new
+      @careplan = if @patient.pctps.in_progress.exists?
+        @patient.pctps.in_progress.first
+      else
+        @patient.pctps.create!(user: current_user)
+      end
+      redirect_to edit_client_health_pctp_careplan_path(@client, @careplan)
+    end
+
+    def edit
+    end
+
+    def update
+      @careplan.assign_attributes(careplan_params)
+      ccm_reviewed = @careplan.reviewed_by_ccm_on.present? && @careplan.reviewed_by_ccm_on_changed?
+      rn_reviewed = @careplan.reviewed_by_rn_on.present? && @careplan.reviewed_by_rn_on_changed?
+
+      @careplan.save
+      @careplan.update(reviewed_by_ccm_id: current_user.id) if ccm_reviewed
+      @careplan.update(reviewed_by_rn_id: current_user.id) if rn_reviewed
+
+      @patient.current_qa_factory.complete_careplan(@careplan) if @careplan.completed?
+      @patient.current_qa_factory.review_careplan(@careplan) if @careplan.reviewed?
+      @patient.current_qa_factory.approve_careplan(@careplan) if @careplan.approved?
```

Comment:
Haven't gotten there yet, but noting that saving a care plan might generate a bunch of these if you save it multiple times...  (Maybe it's guarded by already having one.)

Reply:
This is guarded from duplication.